src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,405 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Results\DTO;
+
+use WordPress\AiClient\Files\Contracts\FileInterface;
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\Enums\MessagePartTypeEnum;
+use WordPress\AiClient\Results\Contracts\ResultInterface;
+
+/**
+ * Represents the result of a generative AI operation.
+ *
+ * This DTO contains the generated candidates along with usage statistics
+ * and metadata from the AI provider.
+ *
+ * @since n.e.x.t
+ */
+class GenerativeAiResult implements ResultInterface
+{
+    /**
+     * @var string Unique identifier for this result.
+     */
+    private string $id;
+
+    /**
+     * @var Candidate[] The generated candidates.
+     */
+    private array $candidates;
+
+    /**
+     * @var TokenUsage Token usage statistics.
+     */
+    private TokenUsage $tokenUsage;
+
+    /**
+     * @var array<string, mixed> Provider-specific metadata.
+     */
+    private array $providerMetadata;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $id Unique identifier for this result.
+     * @param Candidate[] $candidates The generated candidates.
+     * @param TokenUsage $tokenUsage Token usage statistics.
+     * @param array<string, mixed> $providerMetadata Provider-specific metadata.
+     * @throws \InvalidArgumentException If no candidates provided.
+     */
+    public function __construct(string $id, array $candidates, TokenUsage $tokenUsage, array $providerMetadata = [])
+    {
+        if (empty($candidates)) {
+            throw new \InvalidArgumentException('At least one candidate must be provided');
+        }
+
+        $this->id = $id;
+        $this->candidates = $candidates;
+        $this->tokenUsage = $tokenUsage;
+        $this->providerMetadata = $providerMetadata;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Gets the generated candidates.
+     *
+     * @since n.e.x.t
+     *
+     * @return Candidate[] The candidates.
+     */
+    public function getCandidates(): array
+    {
+        return $this->candidates;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getTokenUsage(): TokenUsage
+    {
+        return $this->tokenUsage;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @since n.e.x.t
+     */
+    public function getProviderMetadata(): array
+    {
+        return $this->providerMetadata;
+    }
+
+    /**
+     * Gets the total number of candidates.
+     *
+     * @since n.e.x.t
+     *
+     * @return int The total number of candidates.
+     */
+    public function getTotalCandidates(): int
```

Comment:
Maybe call this `getCandidateCount()`? This would match the `CANDIDATE_COUNT` model option, which controls how many candidates to generate.

Reply:
Resolved in c5b632e